src/services/lease-service/tests/index.test.ts

```diff
@@ -488,4 +490,185 @@ describe('lease-service', () => {
       expect(tenantLeaseAdapter.isLeaseActive(lease)).toBe(false)
     })
   })
+
+  describe('GET applicants/validateResidentialAreaRentalRules/:contactCode/:listingId', () => {
+    it('responds with 404 if listing does not exist', async () => {
+      const getListingSpy = jest
+        .spyOn(listingAdapter, 'getListingById')
+        .mockResolvedValue(undefined)
+
+      const res = await request(app.callback()).get(
+        `/applicants/validateResidentialAreaRentalRules/123/456}`
+      )
+
+      expect(getListingSpy).toHaveBeenCalled()
+      expect(res.status).toBe(404)
+      expect(res.body.reason).toBe('Listing was not found')
+    })
+
+    it('responds with 404 if applicant does not exist', async () => {
+      const listing = ListingFactory.params({
+        districtCode: 'OXB',
+      }).build()
+
+      const getListingSpy = jest
+        .spyOn(listingAdapter, 'getListingById')
+        .mockResolvedValue(listing)
```

Comment:
Skulle säga använd .mockResolvedValueOnce överallt när du kör spy returns inuti enskilda tester för att inte riskera att det läcker in i andra tester.